nncf/quantization/algorithms/weight_compression/openvino_backend.py

```diff
@@ -190,6 +195,69 @@ def dump_parameters(
     ) -> None:
         dump_parameters(model, parameters, algo_name, path)
 
+    @staticmethod
+    def get_compress_decompress_pipeline(
+        weight_compression_parameter: WeightCompressionParameters, w_shape, s_shape, z_p_shape
+    ):
+        config = weight_compression_parameter.compression_config
+        mode = config.mode
+        assert mode in [CompressWeightsMode.INT4_SYM, CompressWeightsMode.INT4_ASYM]
+        num_bits = config.num_bits
+
+        level_low = 0
+        level_high = 2**num_bits - 1
+
+        input_node_w = opset.parameter(w_shape, name="w")
+        input_node_s = opset.parameter(s_shape, name="s")
+        input_node_zp = opset.parameter(z_p_shape, name="zp")
+
+        node_compression_div = opset.divide(input_node_w, input_node_s)
+        node_compression_add = opset.add(node_compression_div, input_node_zp)
+        node_compression_round = opset.round(node_compression_add)
+        node_compression_clamp = opset.clamp(node_compression_round, level_low, level_high)
+
+        result1 = opset.result(node_compression_clamp, name="compressed_weights")
+        result1.get_output_tensor(0).set_names(set(["compressed_weights"]))
+
+        node_decompression_add = opset.subtract(node_compression_clamp, input_node_zp)
+        node_decompression_mul = opset.multiply(node_decompression_add, input_node_s)
+        result2 = opset.result(node_decompression_mul, name="q_weights")
+        result2.get_output_tensor(0).set_names(set(["q_weights"]))
+
+        model = ov.Model([result1, result2], [input_node_w, input_node_s, input_node_zp])
+
+        compiled_model = ov.compile_model(model)
+
+        return compiled_model
+
+    @staticmethod
+    def get_compress_pipeline(weight_compression_parameter: WeightCompressionParameters, w_shape, s_shape, z_p_shape):
+        config = weight_compression_parameter.compression_config
+        mode = config.mode
+        assert mode in [CompressWeightsMode.INT4_SYM, CompressWeightsMode.INT4_ASYM]
```

Comment:
Looks like some code is duplicating over here, can we put the compress/ decompress node creation in separate functions?

Reply:
Done